src/package/packagePushUpgrade.ts

```diff
@@ -128,38 +134,50 @@ export class PackagePushUpgrade {
     packageVersionId: string,
     scheduleTime: string,
     orgList: string[]
-  ): Promise<PackagePushScheduleResult> {
+  ): Promise<PackagePushScheduleResult | SfError> {
     try {
-      const pushRequest = await connection.tooling.create('PackagePushRequest', {
+      const packagePushRequestBody = {
         PackageVersionId: packageVersionId,
         ScheduledStartTime: scheduleTime,
-      });
-
-      if (!pushRequest.success) {
-        throw new Error('Failed to create PackagePushRequest');
-      }
+      };
 
-      // Create PackagePushJob for each org using Bulk API v2
-      const job = connection.bulk2.createJob({ object: 'PackagePushJob', operation: 'insert' });
+      const pushRequestResult: PackagePushRequestResult = await connection.request({
+        method: 'POST',
+        url: `/services/data/v${connection.getApiVersion()}/sobjects/packagepushrequest/`,
+        body: JSON.stringify(packagePushRequestBody),
+      });
 
       const pushJobs = orgList.map((orgId) => ({
-        PackagePushRequestId: pushRequest.id,
+        PackagePushRequestId: pushRequestResult.id,
         SubscriberOrganizationKey: orgId,
       }));
 
-      await job.check();
+      // Create PackagePushJob for each org using Bulk API v2
+      const job = connection.bulk2.createJob({ operation: 'insert', object: 'PackagePushJob' });
+
+      await job.open();
+
       await job.uploadData(pushJobs);
       await job.close();
+      await job.poll();
 
       // If there are any errors for a job, write all specific job errors to an output file
       const jobErrors = await job.getFailedResults();
 
       if (jobErrors.length > 0) {
-        await this.writeJobErrorsToFile(pushRequest.id, jobErrors);
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const error: SfError = await this.writeJobErrorsToFile(pushRequestResult?.id, jobErrors);
+        return error;
```

Comment:
you should avoid passing errors as values through functions unless, it's very easy for callers to ignore these.
suggestion:
make `writeJobErrorsToFile` return the filePath that was written and build & throw the error here, `sf` will catch the error and finish the process.